src/types/ajv/CycleRecordSchema.ts

```diff
@@ -0,0 +1,366 @@
+                      dataType: { type: 'string' },
+                      value: { type: 'string' },
+                    },
+                    required: ['dataType', 'value'],
+                  },
+                },
+                required: ['certExp', 'nominator', 'nominee', 'signs', 'stake'],
+              },
+              version: { type: 'string' },
+            },
+            required: ['mustUseAdminCert', 'stakeCert', 'version'],
+          },
+          cycleMarker: { type: 'string' },
+          nodeInfo: {
+            type: 'object',
+            properties: {
+              publicKey: { type: 'string' },
+              externalIp: { type: 'string' },
+              externalPort: { type: 'number' },
+              internalIp: { type: 'string' },
+              internalPort: { type: 'number' },
+              joinRequestTimestamp: { type: 'number' },
+              readyTimestamp: { type: 'number' },
+              address: { type: 'string' },
+              activeTimestamp: { type: 'number' },
+              syncingTimestamp: { type: 'number' },
+              activeCycle: { type: 'number' },
+              refreshedCounter: { type: 'number' },
+            },
+            required: [
+              'publicKey',
+              'externalIp',
+              'externalPort',
+              'internalIp',
+              'internalPort',
+              'activeTimestamp',
+              'syncingTimestamp',
+              'activeCycle',
+              'joinRequestTimestamp',
+              'address',
+            ],
+          },
+          proofOfWork: { type: 'string' },
+          selectionNum: { type: 'string' },
+          sign: {
+            type: 'object',
+            properties: {
+              owner: { type: 'string' },
+              sig: { type: 'string' },
+            },
+            required: ['owner', 'sig'],
+          },
+          version: { type: 'string' },
+        },
+        required: [
+          'appJoinData',
+          'cycleMarker',
+          'nodeInfo',
+          'proofOfWork',
+          'selectionNum',
+          'sign',
+          'version',
+        ],
+      },
+    },
+    standbyNodeListHash: { type: 'string' },
+    standbyRefresh: { type: 'array', items: { type: 'string' } },
+    standbyRemove: { type: 'array', items: { type: 'string' } },
+    startedSyncing: { type: 'array', items: { type: 'string' } },
+    syncing: { type: 'number' },
+    target: { type: 'number' },
+    desired: { type: 'number' },
+    random: { type: 'number' },
+    networkDataHash: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          cycle: { type: 'number' },
+          hash: { type: 'string' },
+        },
+      },
+    },
+    networkReceiptHash: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          cycle: { type: 'number' },
+          hash: { type: 'string' },
+        },
+      },
+    },
+    networkSummaryHash: {
```

Comment:
networkSummaryHash needs to be required field

Reply:
not required in practice